lib/manageiq_performance/middlewares/memory.rb

```diff
@@ -0,0 +1,55 @@
+module ManageIQPerformance
+  module Middlewares
+    module Memory
+      private
+
+      def memory_initialize
+      end
+
+      def memory_start(env)
+        Memory.memory_query_data = memory_parse_data
+      end
+
+      def memory_finish(env)
+        Memory.memory_query_data = memory_parse_data(Memory.memory_query_data)
+        save_report env, :memory,
+                    memory_short_form_data,
+                    memory_long_form_data
+      ensure
+        Memory.memory_query_data = nil
+      end
+
+      def memory_long_form_data
+        proc { Memory.memory_query_data }
+      end
+
+      alias_method memory_long_form_data memory_short_form_data
```

Comment:
While cute, I think this is causing your test failures.  Defining it manually seems to work just fine (if you want to be DRY about it, maybe have `short_form` just call the `long_form`?).